src/lib/gossip_net/gossip_net.ml

```diff
@@ -238,41 +237,6 @@ module Make (Message : Message_intf) : S with type msg := Message.msg = struct
   let is_unix_errno errno unix_errno =
     Int.equal (Unix.Error.compare errno unix_errno) 0
 
-  let get_socket_reader_writer t peer =
-    (* Create a socket bound to the correct IP, for outgoing connections. *)
-    let socket =
-      let unbound_socket = Async.Socket.(create Type.tcp) in
-      (* Binding with a source port of 0 tells the kernel to pick a free one for
-       us. Because we're binding an address and port before the kernel knows
-       whether this will be a listen socket or an outgoing socket, it won't
-       allocate the same source port twice, even when the destination IP will be
-       different. So we could in very extreme cases run out of ephemeral ports.
-       The IP_BIND_ADDRESS_NO_PORT socket option informs the kernel we're
-       binding the socket to an IP and intending to make an outgoing connection,
-       but it only works on Linux. I think we don't need to worry about it,
-       there are a lot of ephemeral ports and we don't make very many
-       simultaneous connections.
-      *)
-      Async.Socket.bind_inet unbound_socket
-      @@ `Inet (t.addrs_and_ports.bind_ip, 0)
-    in
-    let peer_addr = `Inet (peer.Peer.host, peer.communication_port) in
-    try_with (fun () ->
-        let%map connected_socket = Async.Socket.connect socket peer_addr in
```

Comment:
I guess this would throw, leading to the leak?

Reply:
yeah.